### PR TITLE
FEATURE: RAIL-2140 Include ExampleWithExport code on live examples page

### PR DESCRIPTION
**Pull request by Tulean**
[Check PR owner responsibilities](https://confluence.intgdc.com/display/Development/Code-reviews#Code-reviews-Ownerresponsibilities)

Supported PR commands:

Command | Description
--- | ---
`extended test - examples` | Live examples tests
`extended test - storybook` | Storybook screenshot tests

See more [options](https://confluence.intgdc.com/display/kbhr/How+to+work+with+git+and+Github#HowtoworkwithgitandGithub-Extendedchecks).

# PR Checklist

- [x] Verify code changes ([Checklist](https://confluence.intgdc.com/display/Development/Code-reviews+checklist), [Best practices](https://confluence.intgdc.com/display/Development/Code-reviews+best+practices))
- [x] [Verify pull-request formalities](https://confluence.intgdc.com/display/Development/Code-reviews)
- [x] Change was tested by using [gdc-dev-release](https://confluence.intgdc.com/display/~tomas.vojtasek/Private+NPM) in [gdc-analytical-designer](https://github.com/gooddata/gdc-analytical-designer) and [gdc-dashboards](https://github.com/gooddata/gdc-dashboards) (if applicable)
- [x] Migration guide (for major changes) is mentioned in [CHANGELOG.md](../blob/master/CHANGELOG.md).
- [x] Successful `extended test - examples`
- [x] Successful `extended test - storybook`
- [x] Checked yarn.lock consistency (no dep. duplicities especially Goodstrap)

# Related PRs

- gdc-analytical-designer:
- gdc-dashboards:

# Related Jira tasks

### CODE REVIEW

**Comment by Tulean** [comment]
extended test - storybook

**Comment by gdgate** [comment]
Build succeeded (check pipeline).

- https://checklist.intgdc.com/job/client-libs/job/gooddata-react-components-unit-tests-zuul-docker-el7/6495/ : SUCCESS in 9m 22s
- https://checklist.intgdc.com/job/compliance/job/gooddata-react-components-oss-check/2291/ : SUCCESS in 14s

**Comment by yenkins** [comment]
Link to Storybook test result:

https://checklist.intgdc.com/job/client-libs/job/gooddata-react-components-storybook-zuul-docker/3372/artifact/test-storybook/html-report/index.html

**Comment by gdgate** [comment]
Build succeeded (check-extended pipeline).

- https://checklist.intgdc.com/job/compliance/job/gooddata-react-components-oss-check/2292/ : SUCCESS in 30s
- https://checklist.intgdc.com/job/client-libs/job/gooddata-react-components-pull-request-dispatcher/2774/ : SUCCESS in 17m 52s

**Comment by Tulean** [comment]
extended test - examples

**Comment by yenkins** [comment]
:arrow_right: Started Live Examples deploy to:
https://gdc-live-examples-1444.herokuapp.com

It will be destroyed on merge/close. See [ci-infra](https://github.com/gooddata/ci-infra/blob/master/jenkins/jobs/templates/orchestrators/client-pr-dispatcher.groovy) and [gdc-ci.yaml](https://github.com/gooddata/gooddata-react-components/blob/master/gdc-ci.yaml).

**Comment by gdgate** [comment]
Build succeeded (check-extended pipeline).

- https://checklist.intgdc.com/job/compliance/job/gooddata-react-components-oss-check/2295/ : SUCCESS in 8s
- https://checklist.intgdc.com/job/client-libs/job/gooddata-react-components-pull-request-dispatcher/2777/ : SUCCESS in 20m 00s

**Comment by gdgate** [comment]
Build in pipeline check aborted.

**Comment by gdgate** [comment]
Build in pipeline check aborted.

File: examples/src/components/utils/ExampleWithExport.jsx [hunk 1]
```diff
@@ -14,13 +14,19 @@ export class ExampleWithExport extends React.Component {
...
             showExportDialog: false,
             errorMessage: null,
+            exportResult: undefined,
         };
 
         this.doExport = this.doExport.bind(this);
     }
 
+    /**
+     * This will be called by child visualozation once it successfully renders
```
Comment (no23reason) [inline examples/src/components/utils/ExampleWithExport.jsx]: Typo: child visualization

File: examples/src/components/utils/ExampleWithExport.jsx [hunk 2]
```diff
@@ -14,13 +14,19 @@ export class ExampleWithExport extends React.Component {
...
+     * and its underlying data is ready for export. The child visualization will
+     * call this with one parameter: function that can bre called to trigger the exports.
```
Comment (no23reason) [inline examples/src/components/utils/ExampleWithExport.jsx]: Typo: be called

File: examples/src/components/utils/ExampleWithExport.jsx [hunk 3]
```diff
@@ -14,13 +14,19 @@ export class ExampleWithExport extends React.Component {
...
+     */
     onExportReady = exportResult => {
-        this.exportResult = exportResult;
+        this.setState({ exportResult });
```
Comment (no23reason) [inline examples/src/components/utils/ExampleWithExport.jsx]: Why was the move to state necessary?

**Comment by no23reason** [comment]
extended test - examples

**Comment by yenkins** [comment]
:arrow_right: Started Live Examples deploy to:
https://gdc-live-examples-1444.herokuapp.com

It will be destroyed on merge/close. See [ci-infra](https://github.com/gooddata/ci-infra/blob/master/jenkins/jobs/templates/orchestrators/client-pr-dispatcher.groovy) and [gdc-ci.yaml](https://github.com/gooddata/gooddata-react-components/blob/master/gdc-ci.yaml).

**Comment by gdgate** [comment]
Build succeeded (check pipeline).

- https://checklist.intgdc.com/job/client-libs/job/gooddata-react-components-unit-tests-zuul-docker-el7/6498/ : SUCCESS in 10m 07s
- https://checklist.intgdc.com/job/compliance/job/gooddata-react-components-oss-check/2298/ : SUCCESS in 5s

File: examples/src/components/utils/ExampleWithExport.jsx (hunk 3 above)
Comment (Tulean) [reply examples/src/components/utils/ExampleWithExport.jsx]: This was suggested by lubko in order to sanitize the code. exportResult is a function that is available after a successful render of the child. Therefore, it should in the state.

**Comment by gdgate** [comment]
Build in pipeline check-extended aborted.

**Comment by Tulean** [comment]
extended test - examples

**Comment by yenkins** [comment]
:arrow_right: Started Live Examples deploy to:
https://gdc-live-examples-1444.herokuapp.com

It will be destroyed on merge/close. See [ci-infra](https://github.com/gooddata/ci-infra/blob/master/jenkins/jobs/templates/orchestrators/client-pr-dispatcher.groovy) and [gdc-ci.yaml](https://github.com/gooddata/gooddata-react-components/blob/master/gdc-ci.yaml).

File: examples/src/components/utils/ExampleWithExport.jsx (hunk 3 above)
Comment (no23reason) [reply examples/src/components/utils/ExampleWithExport.jsx]: OK, thanks :)

**Comment by gdgate** [comment]
Build in pipeline check aborted.

**Comment by gdgate** [comment]
Build succeeded (check-extended pipeline).

- https://checklist.intgdc.com/job/compliance/job/gooddata-react-components-oss-check/2302/ : SUCCESS in 5s
- https://checklist.intgdc.com/job/client-libs/job/gooddata-react-components-pull-request-dispatcher/2779/ : SUCCESS in 25m 20s

**Comment by gdgate** [comment]
Build succeeded (gate pipeline).

- https://checklist.intgdc.com/job/client-libs/job/gooddata-react-components-storybook-zuul-docker/3376/ : SUCCESS in 18m 00s
- https://checklist.intgdc.com/job/client-libs/job/gooddata-react-components-unit-tests-zuul-docker-el7/6503/ : SUCCESS in 8m 41s
- https://checklist.intgdc.com/job/compliance/job/gooddata-react-components-merge-compliance-check/1567/ : SUCCESS in 22s
- https://checklist.intgdc.com/job/client-libs/job/gooddata-react-components-examples-zuul-docker/2253/ : SUCCESS in 25m 10s

**Pull request merged**